### PR TITLE
Fix deprecations in examples

### PR DESCRIPTION
**Pull request by QuLogic**
## PR Summary

This fixes several deprecation warnings that are triggered by the examples.

Currently, `plt.plotfile` calls `mlab.csv2rev`, which is deprecated. Because that function provides some automatic converters, I could not replace it with `np.genfromtxt` or `np.loadtxt` right away, as that would be a behaviour change. Even the examples take advantage of this behaviour.

`plt.plotfile` also triggers a warning about re-using an `Axes` spec. I have not looked too deeply into fixing that one.

## PR Checklist

- [x] Has Pytest style unit tests
- [x] Code is PEP 8 compliant
- [N/A] New features are documented, with examples if plot related
- [N/A] Documentation is sphinx and numpydoc compliant
- [N/A] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [N/A] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by QuLogic** [comment]
There are also a few warnings about `tight_layout` which I don't know how to fix; maybe @jklymak has an idea.

File: examples/ticks_and_spines/date_index_formatter.py [hunk 1]
```diff
@@ -12,15 +12,19 @@
...
 import matplotlib.pyplot as plt
 import matplotlib.cbook as cbook
+from matplotlib.dates import bytespdate2num, num2date
 from matplotlib.ticker import Formatter
 
+
 datafile = cbook.get_sample_data('msft.csv', asfileobj=False)
 print('loading %s' % datafile)
-r = csv2rec(datafile)[-40:]
+r = np.genfromtxt(datafile, delimiter=',', names=True,
```
Comment (timhoffm) [inline examples/ticks_and_spines/date_index_formatter.py]: Maybe use a more telling name like `msft_data` instead of `r`?

**Comment by jklymak** [comment]
Oh, hmmm, that should give a warning, but for constrained layout, not tight layout!  Ooops..

```
lib/matplotlib/tests/test_constrainedlayout.py::test_constrained_layout7[png]
  /home/travis/build/matplotlib/matplotlib/lib/matplotlib/figure.py:2269: UserWarning: This figure includes Axes that are not compatible with tight_layout, so results might be incorrect.
    warnings.warn("This figure includes Axes that are not compatible "
```

**Comment by QuLogic** [comment]
@jklymak Actually, I was referring to the examples, not the tests. For example,
```
generating gallery for gallery/lines_bars_and_markers... [ 97%] markevery_demo.py
/home/circleci/project/lib/matplotlib/figure.py:2269: UserWarning: This figure includes Axes that are not compatible with tight_layout, so results might be incorrect.
  warnings.warn("This figure includes Axes that are not compatible "
generating gallery for tutorials/intermediate... [ 83%] tight_layout_guide.py
/home/circleci/project/lib/matplotlib/gridspec.py:316: UserWarning: This figure includes Axes that are not compatible with tight_layout, so results might be incorrect.
  warnings.warn("This figure includes Axes that are not compatible "
generating gallery for tutorials/intermediate... [100%] constrainedlayout_guide.py
/home/circleci/project/lib/matplotlib/figure.py:2224: UserWarning: Calling figure.constrained_layout, but figure not setup to do constrained layout.     You either called GridSpec without the fig keyword, you are using plt.subplot, or you need to call figure or subplotswith the constrained_layout=True kwarg.
  warnings.warn("Calling figure.constrained_layout, but figure "
```

**Comment by jklymak** [comment]
The last one is mine, and on purpose to show the failure in the tutorial.  

The `tight_layout` example is similar, showing that tight_layout doesn't work and emits a warning for colorbars created outside of gridspec. Not sure about the first one.

File: lib/matplotlib/pyplot.py [hunk 2]
```diff
@@ -2411,8 +2411,11 @@ def plotfile(fname, cols=(0,), plotfuncs=None,
 
     if plotfuncs is None:
         plotfuncs = dict()
-    r = mlab.csv2rec(fname, comments=comments, skiprows=skiprows,
-                     checkrows=checkrows, delimiter=delimiter, names=names)
+    from matplotlib.cbook import mplDeprecation
```
Comment (dstansby) [inline lib/matplotlib/pyplot.py]: Should this whole method be deprecated since it depends on `csv2rec`?

Comment (tacaswell) [reply lib/matplotlib/pyplot.py]: Yes, and point people at pandas.

Comment (QuLogic) [reply lib/matplotlib/pyplot.py]: You want to deprecate `plt.plotfile` for Pandas? Probably should be an independent PR.

**Pull request merged**